### PR TITLE
Move Calendar and DatePicker to legacy package

### LINKED ISSUE #9245: Move Calendar and DatePicker to legacy.
~Export date-time versions from OUFR~ (Can't do because of circular dependency)

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9245
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Move the old Calendar and DatePicker components to the legacy package. Also fixed API doc generation for the legacy package.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9253)

### CODE REVIEW

File: packages/api-docs/src/generatePageJsonFiles.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import * as path from 'path';
 import { generateJson } from './PageJsonGenerator';
 
-// Generate JSON for office-ui-fabric-react, styling, utilities, and merge-styles
+// Generate JSON for office-ui-fabric-react, styling, utilities, merge-styles, and legacy
```
Comment (ecraig12345) [inline packages/api-docs/src/generatePageJsonFiles.ts]: @natalieethell This seems to work okay but I'd appreciate if you could take a look

**Comment by kenotron** [comment]
I think you might need to update some things on the perf-test? Some of these things are blank.

**Comment by ecraig12345** [comment]
Given that I didn't touch buttons, I doubt those are my fault. Maybe it's related to @khmakoto's recent PR touching Button in Fabric 7?

**Comment by khmakoto** [comment]
Yes, I think those Button's are related to my PR. I'll check on why those entries are appearing as blanks.

**Comment by kenotron** [comment]
Chatted offline with @jdhuntington about these deprecation. Was going to comment about lacking code mods but realized those are all separate work items. So that's fine! I was also hoping that there would be a dedicated "legacy" section in the website for these components so people know not to choose them if they can help it -also another work item :)

**Comment by ecraig12345** [comment]
@natalieethell Oops, I checked the Implementation sections for Calendar/DatePicker/Nav but assumed everything else would still work... Thanks for catching that!

**Comment by natalieethell** [comment]
Looks like these lines are the culprit. 

https://github.com/OfficeDev/office-ui-fabric-react/blob/0c51f3d0187279e5a86f011281035db51a7e56ef/packages/api-docs/src/PageJsonGenerator.tsx#L97-L99

Since this is in the following for loop

https://github.com/OfficeDev/office-ui-fabric-react/blob/0c51f3d0187279e5a86f011281035db51a7e56ef/packages/api-docs/src/PageJsonGenerator.tsx#L85

the page data associated with each page name is being overwritten with the addition of the legacy apiJsonPath. This isn't an issue for references, because there aren't any pageNames in that blob:

https://github.com/OfficeDev/office-ui-fabric-react/blob/0c51f3d0187279e5a86f011281035db51a7e56ef/packages/api-docs/src/generatePageJsonFiles.ts#L14

**Comment by natalieethell** [comment]
Okay here's what I think will work:

We can move the following lines 

https://github.com/OfficeDev/office-ui-fabric-react/blob/0c51f3d0187279e5a86f011281035db51a7e56ef/packages/api-docs/src/PageJsonGenerator.tsx#L96-L99

out of the for loop, so right above line 85

https://github.com/OfficeDev/office-ui-fabric-react/blob/0c51f3d0187279e5a86f011281035db51a7e56ef/packages/api-docs/src/PageJsonGenerator.tsx#L85

The `pageNames` object should be the same for each of the `apiJsonPath`s, so it doesn't need to happen inside the for loop.

**Pull request merged**